### ISSUE TITLE
Pandas is not found by Data Viewer

### ISSUE THREAD

**Issue by juanfcocontreras** [microsoft/vscode-jupyter#9921]
### Applies To

- [ ] Notebooks (.ipynb files)
- [X] Interactive Window and\/or Cell Scripts (.py files with \#%% markers)

### What happened?

Conda environment is not used anymore to get pandas but system interpreter is, while I am debugging a Python script. It is still working in Notebooks, but not in regular scripts in my case. It didn't happen in previouse versions.

### VS Code Version

Version: 1.67.0 Commit: 57fd6d0195bb9b9d1b49f6da5db789060795de47 Date: 2022-05-04T12:14:47.100Z Electron: 17.4.1 Chromium: 98.0.4758.141 Node.js: 16.13.0 V8: 9.8.177.13-electron.0 OS: Darwin x64 19.6.0

### Jupyter Extension Version

2022.5.1001281006

### Jupyter logs

```shell
Visual Studio Code (1.67.0, undefined, desktop)
Jupyter Extension Version: 2022.5.1001281006.
Python Extension Version: 2022.6.0.
Workspace folder /Users/user/Desktop/data
info 12:12:29.250: ZMQ install verified.
User belongs to experiment group 'jupyterTestcf'
User belongs to experiment group 'jupyterEnhancedDataViewer'
info 12:12:30.94: Experiment status for python is {"enabled":true,"optInto":[],"optOutFrom":[]}
info 12:12:38.82: Process Execution: > /usr/local/Caskroom/miniconda/base/envs/conda-forge/bin/python -m pip list
> /usr/local/Caskroom/miniconda/base/envs/conda-forge/bin/python -m pip list
info 12:12:59.704: Process Execution: > /usr/bin/python -c "import pandas;print(pandas.__version__)"
> /usr/bin/python -c "import pandas;print(pandas.__version__)"
warn 12:12:59.757: Failed to get version of Pandas to use Data Viewer [i [Error]: Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named pandas

	at ChildProcess.<anonymous> (/Users/user/.vscode/extensions/ms-toolsai.jupyter-2022.5.1001281006/out/extension.node.js:2:836693)
	at Object.onceWrapper (node:events:510:26)
	at ChildProcess.emit (node:events:390:28)
	at maybeClose (node:internal/child_process:1064:16)
	at Process.ChildProcess._handle.onexit (node:internal/child_process:301:5)] {
  category: 'unknown'
}
error 12:13:1.243: [Error: Python package 'pandas' is required for viewing data.
	at b.installMissingDependencies (/Users/user/.vscode/extensions/ms-toolsai.jupyter-2022.5.1001281006/out/extension.node.js:2:1211784)
	at processTicksAndRejections (node:internal/process/task_queues:96:5)
	at async b.checkAndInstallMissingDependencies (/Users/user/.vscode/extensions/ms-toolsai.jupyter-2022.5.1001281006/out/extension.node.js:2:1211157)
	at async x.onVariablePanelShowDataViewerRequest (/Users/user/.vscode/extensions/ms-toolsai.jupyter-2022.5.1001281006/out/extension.node.js:2:57932)
	at async o._executeContributedCommand (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:83:31325)]
warn 12:13:1.244: DataScience Error [Error: Python package 'pandas' is required for viewing data.
	at b.installMissingDependencies (/Users/user/.vscode/extensions/ms-toolsai.jupyter-2022.5.1001281006/out/extension.node.js:2:1211784)
	at processTicksAndRejections (node:internal/process/task_queues:96:5)
	at async b.checkAndInstallMissingDependencies (/Users/user/.vscode/extensions/ms-toolsai.jupyter-2022.5.1001281006/out/extension.node.js:2:1211157)
	at async x.onVariablePanelShowDataViewerRequest (/Users/user/.vscode/extensions/ms-toolsai.jupyter-2022.5.1001281006/out/extension.node.js:2:57932)
	at async o._executeContributedCommand (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:83:31325)]
```

**Comment by roderpad** [comment]
The same thing is happening to me as well. I use the Data Viewer to view dataframes when I'm debugging. Everything was working fine until I updated VSCode yesterday (now using same version/commit as user above).  The problem is also exactly the same, where initially loading the terminal it shows that conda env python is being used. For me the logs show "Process Execution: > **_~\Anaconda3\envs\conda-env-name\python.exe_**" which is correct, but the error pops up when I right-click on a local variable while debugger is active at a breakpoint and select 'View Value in Data Viewer', where the error is 'Process Execution: > **_c:\Program Files\Python36\python.exe_** -c "import pandas;print(pandas.__version__)". Then it throws the "Failed to get version of Pandas to use Data Viewer [i [Error]: .... No module named 'pandas'.". 

Sorry I can't post any screenshots, as it's on my work computer, but I just wanted to mention I am also having the issue and it was working before the VSCode update.

**Comment by amunger** [comment]
Thanks for reporting the issue, I haven't managed to repro yet, but maybe I don't quite understand the setup.

Do you have the conda environment selected as your active interpreter? should show up in the lower right when you open a .py file.
Are you changing the selected kernel in the Interactive window, or just leaving it as the active interpreter?

**Comment by roderpad** [comment]
Yes, I have the conda env selected as the active interpreter. Typically, if I were working in a notebook, I would change the kernel to be sure it is using the conda env; but I am not using a notebook. Rather, I believe the Data Viewer in the Debug console uses Jupyter to view dataframes. However, when looking at the Jupyter logs while I run the debugger, it shows that it begins using my conda env, but when I select 'View Value in Data Viewer' it uses the wrong interpreter. 

Yesterday I tried things like opening a notebook and setting the Kernel to the conda env, hoping it would force the jupyter extension to use that during debugging, but that didn't work. I also tried uninstalling and reinstalling python, pylance, and jupyter vscode extensions, to no avail. 

I'm still able to do everything else just fine (run my programs, debug, etc), I just can't use the 'View Value in Data Viewer' option anymore, as it throws the 'Pandas not found' error.
